programs/whirlpool/src/instructions/open_position_with_metadata.rs

```diff
@@ -48,7 +49,9 @@ pub struct OpenPositionWithMetadata<'info> {
     #[account(address = token::ID)]
     pub token_program: Program<'info, Token>,
     pub system_program: Program<'info, System>,
-    pub rent: Sysvar<'info, Rent>,
+    /// CHECK: checked via account constraints
+    #[account(address = sysvar::rent::ID)]
+    pub rent: UncheckedAccount<'info>,
```

Comment:
hmm, could we keep `Sysvar<'info, Rent>` ?

To avoid being given arbitrary addresses.
Not as massively executed as "swap" instruction and does not affect transaction size.
As far as `open_position` is concerned, it should be excluded from `open_position_v2`. As you know, I think `open_position_v2` is clearly preferred because of its advantages (no non-refundable rent overhead).

Reply:
The problem is stack space and not transaction size. See my comment below